docs/source/contributing.md

```diff
@@ -2,154 +2,88 @@
+* [Effective Go](https://go.dev/doc/effective_go)
+* [Go Code Review Comments](https://go.dev/wiki/CodeReviewComments)
+* [API changes](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md)
+* [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md)
 
-Now you are ready to make the changes and commit to your branch.
 
-### Updating Your Fork
+In case the existing code doesn't follow it, try to be consistent.
+Make sure to familiarize yourself with these before preparing a change.
 
-During the development lifecycle, you will need to keep up-to-date with the latest upstream master. As others on the team push changes, you will need to `rebase` your commits on top of the latest. This avoids unnecessary merge commits and keeps the commit history clean.
+## Enhancement proposals
 
-Whenever you need to update your local repository, you never want to merge. You **always** will rebase. Otherwise you will end up with merge commits in the git history. If you have any modified files, you will first have to stash them (`git stash save -u "<some description>"`).
-
-```bash
-git fetch --all
-git rebase upstream/master
-```
-
-Rebasing is a very powerful feature of Git. You need to understand how it works or else you will risk losing your work. Read about it in the [Git documentation](https://git-scm.com/docs/git-rebase), it will be well worth it. In a nutshell, rebasing does the following:
-- "Unwinds" your local commits. Your local commits are removed temporarily from the history.
-- The latest changes from upstream are added to the history
-- Your local commits are re-applied one by one
-- If there are merge conflicts, you will be prompted to fix them before continuing. Read the output closely. It will tell you how to complete the rebase.
-- When done rebasing, you will see all of your commits in the history.
+In case of bigger features or API changes, it's required to prepare an enhancement proposal beforehand as usually these require broader discussions.
+For further details, check https://github.com/scylladb/scylla-operator/tree/master/enhancements.
 
 ## Submitting a Pull Request
 
-Once you have implemented the feature or bug fix in your branch, you will open a PR to the upstream repo. Before opening the PR ensure you have added unit tests, are passing the integration tests, cleaned your commit history, and have rebased on the latest upstream.
-
-In order to open a pull request (PR) it is required to be up to date with the latest changes upstream. If other commits are pushed upstream before your PR is merged, you will also need to rebase again before it will be merged.
+Once you have implemented the feature or bug fix in your branch, you need to open a PR to the upstream repo.
+Before opening the PR ensure you have added unit tests, all e2e's are passing, and your commit history is clean.
```

Comment:
I think it's sometimes valid to test the e2e in CI as a WIP PR - it may be chalenging to setup a working e2e environment, especially with networking or the serial suite.